### ISSUE TITLE
Questions about training

### ISSUE THREAD

**Issue by sunskys** [bobetocalo/bobetocalo_ciarp17#1]
Hello, I’ve been researching CNN-based head pose estimation recently. Your job is very helpful to me. Can the relevant training code in your paper be published? Currently only test code, thanks
At the same time, I think that the head pose estimation is a regression problem, then the original version of resnet is a classification problem. What should we do in the process of using the original network? Thank you

**Comment by bobetocalo** [comment]
I have been unable to train a CNN from C++ code using Caffe yet. You have to do it directly by calling the caffe binary or using the Python interface. 
```
# Create Caffe neural network
import caffe
solver = caffe.get_solver(your_model_solver)
# Train from scratch vs Finetune from a previous net
if transfer_learning:
     solver.net.copy_from(your_model_weights)
solver.solve()
``` 
You have to modify the original https://github.com/bobetocalo/deep_learning/blob/master/CaffeModels/RegressionModels/ResNet_152/ResNet_152_test.prototxt 
```
layer {
	bottom: "pool5"
	top: "fc1000"
	name: "fc1000"
	type: "InnerProduct"
	inner_product_param {
		num_output: 1000
	}
}

layer {
	bottom: "fc1000"
	top: "prob"
	name: "prob"
	type: "Softmax"
}
```
And use as output the InnerProduct result with 3 values (yaw, pitch and roll)
```
layer {
	bottom: "pool5"
	top: "fc_r"
	name: "fc_r"
	type: "InnerProduct"
	inner_product_param {
		num_output: 3
	}
}
```

**Comment by sunskys** [comment]
@bobetocalo  Thank you for your reply. How is the training set produced? Do I only need to change the label at the time of classification to the corresponding three angles? Does it need to modify the caffe file? Thank you

**Comment by bobetocalo** [comment]
The training and test sub sets of AFLW are selected using te attached files according to our paper.
[aflw_ann_train.txt](https://github.com/bobetocalo/bobetocalo_ciarp17/files/1931670/aflw_ann_train.txt)
[aflw_ann_test.txt](https://github.com/bobetocalo/bobetocalo_ciarp17/files/1931671/aflw_ann_test.txt)
We are not allowed to provide the AFLW images.

> Do I only need to change the label at the time of classification to the corresponding three angles? Does it need to modify the caffe file? 
I do not understand these sentences. We have uploaded the trained models in this repository and you can use them to reproduce results or process your own images.
